File: components/birds/core/src/main/kotlin/io/twocan/birds/Bird.kt
package io.twocan.birds

import com.fasterxml.jackson.annotation.JsonProperty
import java.time.Instant
import java.util.*

data class Bird(
    val id: UUID,
    val firstName: String,
    val lastName: String,
    val lastFlight: Instant?,
) {
    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
    val name = "$firstName $lastName"
}

File: frontend-old/src/birds/bird.ts
import {useFetchGet} from '../fetch/fetch-get'

export type Bird = {
    id: string
    name: string
    firstName: string
    lastName: string
    lastFlight: string | null
}

export const useBirds = () => useFetchGet<Bird[]>('/api/birds', 'birds')
